[screengrab] fix screengrab folder name

plastiv opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Fixes https://github.com/fastlane/fastlane/issues/15653#issuecomment-562642978
Fixes #16050

### Description
<!-- Describe your changes in detail. -->
<!-- Please describe in detail how you tested your changes. -->
screengrab companion java library doesn't write to `app_screengrab` folder, but ruby part still tries to read screenshots from there. 

### Testing Steps
<!-- Optional: steps, commands, or code used to test your changes. -->
<!-- Providing these will reduce the time needed for testing and review by the fastlane team. -->
https://github.com/fastlane/fastlane/issues/15653#issuecomment-592065897

plastiv committed:
Fix screengrab folder name

Fixes https://github.com/fastlane/fastlane/issues/15653#issuecomment-562642978

plastiv commented:
@googlebot I signed it!

janpio commented:
Did this work before?
If so, can you pinpoint when this broke?

joshdholtz commented:
Ah, weird! I THINK I see what is going on but running on Android tests to see 🤔

joshdholtz commented on screengrab/lib/screengrab/runner.rb:
```diff
@@ -146,7 +146,7 @@ def determine_internal_screenshots_paths(app_package_name, locales)
         "/data/user/0/#{app_package_name}/files/#{app_package_name}/screengrab/#{locale}/images/screenshots"
       end
```
```suggestion
        [
          "/data/user/0/#{app_package_name}/files/#{app_package_name}/screengrab/#{locale}/images/screenshots",
          
          # https://github.com/fastlane/fastlane/issues/15653#issuecomment-578541663
          "/data/data/#{app_package_name}/files/#{app_package_name}/screengrab/#{locale}/images/screenshots"
        ]
      end.flatten
```

plastiv replied:
Thank you for you time taking to review my PR!

I have checked suggested changes locally and it seems to me this creates duplicates in resolved path. Since `determine_internal_screenshots_paths` unconditionally is called with array of locales. And test run for second locale resolves both first and second locale paths as valid. Example output from emulator 29;google_apis;x86_64:

```shell
21:15:30]: Running tests for locale: de-DE
[21:15:30]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 shell am instrument --no-window-animation -w \
-e testLocale de_DE \
-e endingLocale en_US \
-e appendTimestamp true \
-e package screenshot.mocked \
de.flixbus.app.debug.test/de.meinfernbus.internal.FlixAndroidTestRunner
[21:15:59]: ▸ screenshot.mocked.RidesScreenShooter:.
[21:16:03]: ▸ screenshot.mocked.SearchScreenShooter:.
[21:16:07]: ▸ screenshot.mocked.TicketScreenShooter:.
[21:16:07]: ▸ Time: 31.46
[21:16:07]: ▸ OK (3 tests)
[21:16:08]: Pulling captured screenshots for locale de-DE from the device
ls: /sdcard/de.flixbus.app.debug/screengrab: No such file or directory
[21:16:08]: Exit status: 1
ls: /data/data/de.flixbus.app.debug/app_screengrab: No such file or directory
[21:16:08]: Exit status: 1
ls: /data/data/de.flixbus.app.debug/screengrab: No such file or directory
[21:16:08]: Exit status: 1
[21:16:08]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/user/0/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/de-DE/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-o2knv3
[21:16:08]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/data/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/de-DE/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-o2knv3
ls: /data/user/0/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/cs-CZ/images/screenshots: No such file or directory
[21:16:08]: Exit status: 1
ls: /data/data/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/cs-CZ/images/screenshots: No such file or directory
[21:16:08]: Exit status: 1
[21:16:08]: Screenshots copied to screenshots/de-DE/images/phoneScreenshots
[21:16:08]: Running tests for locale: cs-CZ
[21:16:08]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 shell am instrument --no-window-animation -w \
-e testLocale cs_CZ \
-e endingLocale en_US \
-e appendTimestamp true \
-e package screenshot.mocked \
de.flixbus.app.debug.test/de.meinfernbus.internal.FlixAndroidTestRunner
[21:16:33]: ▸ screenshot.mocked.RidesScreenShooter:.
[21:16:36]: ▸ screenshot.mocked.SearchScreenShooter:.
[21:16:39]: ▸ screenshot.mocked.TicketScreenShooter:.
[21:16:39]: ▸ Time: 16.567
[21:16:39]: ▸ OK (3 tests)
[21:16:39]: Pulling captured screenshots for locale cs-CZ from the device
ls: /sdcard/de.flixbus.app.debug/screengrab: No such file or directory
[21:16:39]: Exit status: 1
ls: /data/data/de.flixbus.app.debug/app_screengrab: No such file or directory
[21:16:40]: Exit status: 1
ls: /data/data/de.flixbus.app.debug/screengrab: No such file or directory
[21:16:40]: Exit status: 1
[21:16:40]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/user/0/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/de-DE/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-7tlqen
[21:16:40]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/data/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/de-DE/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-7tlqen
[21:16:40]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/user/0/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/cs-CZ/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-7tlqen
[21:16:40]: $ /usr/local/share/android-sdk/platform-tools/adb -s emulator-5554 pull /data/data/de.flixbus.app.debug/files/de.flixbus.app.debug/screengrab/cs-CZ/images/screenshots /var/folders/ww/cgv63frx1c5fdzpn183d67wc0000gn/T/d20200303-4970-7tlqen
[21:16:40]: Screenshots copied to screenshots/cs-CZ/images/phoneScreenshots
```
(check last 5 lines. fastlane pulls both de-DE and cs-CZ into screenshots/cs-CZ/images folder)

joshdholtz replied:
Ahhh, this is another issue that we are trying to solve! So this is a good thing you saw this in a way 😊 I'm going to see if I can replicate this on my side 🤔

joshdholtz commented on screengrab/lib/screengrab/runner.rb:
```diff
@@ -146,7 +146,7 @@ def determine_internal_screenshots_paths(app_package_name, locales)
       end
 
-      return ["/data/data/#{app_package_name}/app_screengrab"] + locale_paths
+      return ["/data/data/#{app_package_name}/screengrab"] + locale_paths
```
```suggestion
      return ["/data/data/#{app_package_name}/app_screengrab"]+
                  ["/data/data/#{app_package_name}/screengrab"]+
                  locale_paths
```

joshdholtz reviewed (changes requested):
Can we do something like this?! 😊 This should address another comment in that issue about seeing things in `/data/data` and not in `/data/user/0`

joshdholtz commented:
@plastiv Left some requested changes for you! I tried to make these changes myself and push to this branch but it doesn't look like I have access to do that for some reason 🙃

plastiv committed:
Apply suggestions from code review

Co-Authored-By: Josh Holtz <josh@rokkincat.com>

joshdholtz commented:
@plastiv Can you add... 

```rb
next unless path.include?(locale)
```

inside of the `pull_screenshots_from_device` method? Like in the following screenshot...

![Screen Shot 2020-03-04 at 10 42 08 AM](https://user-images.githubusercontent.com/401294/75901876-cd920380-5e04-11ea-88ad-06861d3bd797.png)

This should fix it 😊

plastiv commented:
> @plastiv Can you add...

Thank you for a detail patch description. I have added requested change and checked locally. It seemed to solve my issue with duplicated folders ✅ 

I have just read the suggested [git flow to address patches](https://github.com/fastlane/fastlane/blob/master/YourFirstPR.md#after-the-review) after I have rebased already. I will create commit-by-commit separately next time.

joshdholtz reviewed (approved):
@plastiv Thanks so much for making these fixes! Really appreciate it ❤️ Hope to see you back here 😊
